pyani_plus/private_cli.py

```diff
@@ -350,28 +318,30 @@ def log_fastani(  # noqa: PLR0913
     if config.configuration_id is None:
         sys.exit("Error with configuration table?")
 
-    sim_errors = fragments - orthologous_matches
-
-    # Need to lookup query length to compute query_cover:
     query_md5 = file_md5sum(query_fasta)
-    query = db_orm.add_genome(session, query_fasta, query_md5)
-    cov_query = float(orthologous_matches) / query.length
-
-    # Need to lookup subject length to compute subject_cover:
     subject_md5 = file_md5sum(subject_fasta)
+
+    # Need to lookup query/subject length to estimate alignment length.
+    # Should not be needed in standard workflow, but also ensures FASTA are in DB:
+    query = db_orm.add_genome(session, query_fasta, query_md5)
     subject = db_orm.add_genome(session, subject_fasta, subject_md5)
-    cov_subject = float(orthologous_matches) / subject.length
+
+    estimated_cov_query = float(orthologous_matches) / fragments  # an approximation
+    sim_errors = fragments - orthologous_matches  # proxy value, not bp
+    estimated_aln_length = int(
```

Comment:
The fragment size used by fastANI is a controllable parameter (default: 3000bp) and reported by fastANI. We could use this directly to report alignment length as total length of orthologous fragments, as `fragment_length` * `orthologous_matches`

Reply:
I like that calculation better than the current estimate 😀

Reply:
For query coverage, we currently have used number of orthologous matched fragments divided by total fragments.

Leighton suggests we _could_ use fragment size (default 3000bp) times number of orthologous fragments, divided by query length. Perhaps an option for later.